nautobot/core/models/constants.py

```diff
@@ -1,11 +1,11 @@
 """Constants for use in Nautobot core models."""
 
-# For our purposes, NATURAL_KEY_SLUG_SEPARATOR needs to be:
-# 1. Safe in a URL path component (so that we can do URLS like "/dcim/devices/<natural_key_slug>/delete/")
+# For our purposes, COMPOSITE_KEY_SEPARATOR needs to be:
+# 1. Safe in a URL path component (so that we can do URLS like "/dcim/devices/<composite_key>/delete/")
 #    Per RFC3986 section 2.3 the general "unreserved" characters are ALPHA and DIGIT and the characters -._~
 #    Per RFC3986 section 3.3 path components also permit characters :@ and the "sub-delims" characters !$&'()*+,;=
 # 2. Not readily confused with characters commonly seen in un-escaped natural key component fields
 #    "." is already ruled out as an unreserved character but also would appear in IPv4 IPAddress and Prefix objects
 #    ":" similarly would appear in IPv6 IPAddress/Prefix objects
 #    "/" would appear in Prefix objects as well as various numbered device component names
-NATURAL_KEY_SLUG_SEPARATOR = "&"
+COMPOSITE_KEY_SEPARATOR = "&"
```

Comment:
Not blocking but let's think about revisiting this in the very near future - I'm just thinking that `&` is going to be problematic in query strings if/when we want NaturalKeyOrPKMultipleChoiceFilter to support composite-keys (`/dcim/devices/?location=Campus&Location1` will not work). Maybe we should change this to `!` or `;`? Needs some thought.

Reply:
Changing it to `;` per discussion in standup.